src/dfa/LG_Lexer.cpp

```diff
@@ -56,7 +56,7 @@ int   LG_Lexer::get_token ()     // DFA LG_Lexer Engine.
       token.start = token.end;            // Set start of token.
       token.line  = line_numb;            // Set line number of token.
       token.column = col_numb + 1;        // Set column number.
-      while (next = Tm [Tr [state] + Tc [*token.end]])
+      while ((next = Tm [Tr [state] + Tc [*token.end]]))
```

Comment:
Not a fan of assignments in conditionals; it's a source of errors.
But, this is original code,  and the parens are truly not needed.

Reply:
Extra parenthesis is a way to tell the compiler "I know what I'm doing here don't show me a warning".

Reply:
Extra parens are useful to disambiguate some expressions in C-based languages, and some compilers will suggest it.
Emacs can also use extra parens for better syntax-based formatting.
IMO, the much, much better solution for this case is to not have the assignment in the conditional expression; it doesn't generate any better code, but it harder to reason about.